packages/odf/modals/s3-browser/delete-and-empty-bucket/EmptyBucketModal.tsx

```
@@ -0,0 +1,275 @@
+  >;
+};
+
+export const getTextInputLabel = (t: TFunction, bucketName: string) => (
+  <Trans t={t as any} ns="plugin__odf-console">
+    <b>
+      To confirm deletion, type <i>{{ bucketName }}</i> in the text input field.
+    </b>
+  </Trans>
+);
+
+const EmptyBucketModal: React.FC<CommonModalProps<EmptyBucketModalProps>> = ({
+  closeModal,
+  isOpen,
+  extraProps: { bucketName, noobaaS3, refreshTokens, setEmptyBucketResponse },
+}) => {
+  const { t } = useCustomTranslation();
+  const [inputValue, setInputValue] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const onEmpty = async (event) => {
+    event.preventDefault();
+    setInProgress(true);
+
+    try {
+      let isTruncated = true;
+      let keyMarker: string;
+      let versionIdMarker: string;
+
+      while (isTruncated) {
+        const deleteObjectKeys = [];
+        const searchParams = {
+          Bucket: bucketName,
+          KeyMarker: keyMarker,
+          VersionIdMarker: versionIdMarker,
+        };
+
+        // eslint-disable-next-line no-await-in-loop
+        const objects = await noobaaS3.listObjectVersions(searchParams);
+
+        if (objects?.Versions) {
+          deleteObjectKeys.push(
+            ...objects.Versions.map((object) => ({
+              Key: object.Key,
+              VersionId: object.VersionId,
+            }))
+          );
+        }
+        if (objects?.DeleteMarkers) {
+          deleteObjectKeys.push(
+            ...objects.DeleteMarkers.map((marker) => ({
+              Key: marker.Key,
+              VersionId: marker.VersionId,
+            }))
+          );
+        }
+
+        // eslint-disable-next-line no-await-in-loop
+        const deleteResponse = await noobaaS3.deleteObjects({
+          Bucket: bucketName,
+          Delete: { Objects: deleteObjectKeys },
+        });
+
+        setEmptyBucketResponse({
+          response: deleteResponse,
+          bucketName: bucketName,
+        });
```

this is not "early stopping" we discussed... if while deleting objects we get any error in response we need to stop this "while" loop entirely and show the alert on the list page...

here it will keep on looping until the end...

---

ex:
```
if(hasErrors) {
       setEmptyBucketResponse({
          response: deleteResponse,
          bucketName: bucketName,
        });
        throw new Error();
}
```

---

ah right, need to throw the error also, got removed while i was refactoring the errors. putting it back